battleship_oo2_test.py

```diff
@@ -118,17 +115,40 @@ def take_shot(self, shot):
 
 
 Coordinates = NamedTuple('Coordinates', x=int, y=int)
+Shot = NamedTuple('Shot', coordinates=Coordinates, shooter_name=str)
 
 
 class Cell:
     """ This class contains the coordinates of a single cell, and keeps track if it has been hit """
-    def __init__(self, coordinates):
-        self.coordinates = coordinates
-        self.is_hit = False
 
+    def __init__(self, coordinates: 'Coordinates', status: str = None):
+        self._coordinates = coordinates
+        self._status = status
+        self._allowed_statuses = {'Hit', 'Not Hit'}
```

Comment:
Consider using an enum for constants like 'Hit' and 'Not Hit' to eliminate hardcoding these strings multiple times in the code. With an enum you can define valid statuses and use those same enum values across the board.